web/src/lib/components/workflows/SchemaFormFields.svelte

```diff
@@ -0,0 +1,382 @@
+    const currentMetadata = pickerMetadata[key] as (AlbumResponseDto | PersonResponseDto)[];
+
+    updateConfig(
+      key,
+      currentIds.filter((id) => id !== itemId),
+    );
+    pickerMetadata = {
+      ...pickerMetadata,
+      [key]: currentMetadata.filter((item) => item.id !== itemId) as AlbumResponseDto[] | PersonResponseDto[],
+    };
+  };
+
+  const renderPicker = (subType: 'album-picker' | 'people-picker', multiple: boolean) => {
+    const isAlbum = subType === 'album-picker';
+    const handler = isAlbum ? handleAlbumPicker : handlePeoplePicker;
+    const selectSingleLabel = isAlbum ? 'select_album' : 'select_person';
+    const selectMultiLabel = isAlbum ? 'select_albums' : 'select_people';
+
+    const buttonText = multiple ? $t(selectMultiLabel) : $t(selectSingleLabel);
+
+    return { handler, buttonText };
+  };
+</script>
+
+{#snippet pickerItemCard(
+  item: AlbumResponseDto | PersonResponseDto,
+  isAlbum: boolean,
+  size: 'large' | 'small',
+  onRemove: () => void,
+)}
+  {@const sizeClass = size === 'large' ? 'h-16 w-16' : 'h-12 w-12'}
+  {@const textSizeClass = size === 'large' ? 'font-medium' : 'font-medium text-sm'}
+  {@const iconSizeClass = size === 'large' ? 'h-5 w-5' : 'h-4 w-4'}
+  {@const countSizeClass = size === 'large' ? 'text-sm' : 'text-xs'}
+
+  <div
+    class="flex items-center gap-3 rounded-xl border border-gray-200 dark:border-gray-700 bg-white dark:bg-gray-900 p-3 shadow-sm"
+  >
+    <div class="shrink-0">
+      {#if isAlbum && 'albumThumbnailAssetId' in item}
+        {#if item.albumThumbnailAssetId}
+          <img
+            src={getAssetThumbnailUrl(item.albumThumbnailAssetId)}
+            alt={item.albumName}
+            class="{sizeClass} rounded-lg object-cover"
+          />
+        {:else}
+          <div class="{sizeClass} rounded-lg bg-gray-200 dark:bg-gray-700"></div>
+        {/if}
+      {:else if !isAlbum && 'name' in item}
+        <img src={getPeopleThumbnailUrl(item)} alt={item.name} class="{sizeClass} rounded-full object-cover" />
+      {/if}
+    </div>
+    <div class="flex-1 min-w-0">
+      <p class="{textSizeClass} text-gray-900 dark:text-gray-100 truncate">
+        {isAlbum && 'albumName' in item ? item.albumName : 'name' in item ? item.name : ''}
+      </p>
+      {#if isAlbum && 'assetCount' in item}
+        <p class="{countSizeClass} text-gray-500 dark:text-gray-400">
+          {$t('items_count', { values: { count: item.assetCount } })}
+        </p>
+      {/if}
+    </div>
+    <button
+      type="button"
+      onclick={onRemove}
+      class="shrink-0 rounded-full p-1.5 text-gray-500 hover:bg-gray-100 dark:hover:bg-gray-800 transition-colors"
+      aria-label={$t('remove')}
+    >
+      <svg class={iconSizeClass} fill="none" viewBox="0 0 24 24" stroke="currentColor">
+        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M6 18L18 6M6 6l12 12" />
+      </svg>
+    </button>
+  </div>
+{/snippet}
+
+{#snippet pickerField(
```

Comment:
Can you extract this out into a dedicated component, something like `<WorkflowPickerField {component} />`